janus_server/src/datastore.rs

```diff
@@ -485,6 +576,12 @@ pub enum Error {
     /// will never be generated by the datastore library itself.
     #[error(transparent)]
     User(#[from] Box<dyn std::error::Error + Send + Sync>),
+    #[error("URL parse error")]
+    Url(#[from] url::ParseError),
+    #[error("Invalid task parameters: {0}")]
+    TaskParameters(#[from] crate::task::Error),
+    #[error("Integer conversion failed: {0}")]
+    TryFromInt(#[from] std::num::TryFromIntError),
```

Comment:
(very pedantic, sorry)

```suggestion
    #[error("invalid task parameters: {0}")]
    TaskParameters(#[from] crate::task::Error),
    #[error("integer conversion failed: {0}")]
    TryFromInt(#[from] std::num::TryFromIntError),
```

The usual practice with error messages I've seen is to not capitalize them, so you get error log lines that read like "Couldn't frob the widget: couldn't find widget frobber: couldn't access frob database: connection reset by peer" rather than "Couldn't frob the widget: Couldn't find widget frobber: Couldn't access frob database: connection reset by peer." Separately, log lines _should_ start capitalized.

Reply:
You have pointed this out to me three times now, and I promise that this time I will remember it. Thanks!